pkg/controller/clusterprovision/clusterprovision_controller.go

```diff
@@ -272,6 +276,20 @@ func (r *ReconcileClusterProvision) reconcileRunningJob(instance *hivev1.Cluster
 
 	pLog = pLog.WithField("job", job.Name)
 
+	installPod, err := r.getInstallPod(job, pLog)
```

Comment:
This should be done **after** the check to see if the job was successful or failed. If the job is successful or failed, then the pod may not exist, but we want to rely on what the job is telling us happened.